test/metabase/query_processor_test/distinct_where_test.clj

```diff
@@ -0,0 +1,111 @@
+(ns ^:mb/driver-tests metabase.query-processor-test.distinct-where-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase.lib.metadata.jvm :as lib.metadata.jvm]
+   [metabase.lib.test-util :as lib.tu]
+   [metabase.query-processor.store :as qp.store]
+   [metabase.test :as mt]))
+
+(deftest ^:parallel basic-test
+  (mt/test-drivers (mt/normal-drivers-with-feature :distinct-where)
+    (is (= 3
+           (->> {:aggregation [[:distinct-where
+                                [:field (mt/id :venues :price) nil]
+                                [:< [:field (mt/id :venues :price) nil] 4]]]}
```

Comment:
Use the shorthand, `$price`, rather than spelling out the field refs in full.

It's easier to read, but it's also more future-proof if, say, we were to change the spelling of refs!